src/libraries/System.Private.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/Runtime.CS.Owned.cs

```diff
@@ -7,9 +7,9 @@ namespace System.Runtime.InteropServices.JavaScript
 {
     public static partial class Runtime
     {
-        private static readonly Dictionary<int, WeakReference<JSObject>> _csOwnedObjects = new Dictionary<int, WeakReference<JSObject>>();
+        private static readonly Dictionary<IntPtr, WeakReference<JSObject>> _csOwnedObjects = new Dictionary<IntPtr, WeakReference<JSObject>>();
```

Comment:
I think this will significantly regress performance unless you make sure to provide a comparer. IIRC the default dictionary comparer for most types is bad, moving from int to IntPtr probably will hit that.